### PR TITLE
SSL module

### PR DESCRIPTION
**Pull request by gmiu**
New files:
- sample_openssl.py
- openssl.py

### CODE REVIEW

File: hubblestack_nova/modules/openssl.py [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+    notAfter:
+        - the minimum number of days left until the certificate should expire
+        - if the certificate will expire in less then the value given here, the check will fail
+        - if notAfter is missing, the default value is 0; basically, the if the expiration date is in the future, the
+          check will be passed
+    notBefore:
+        - the expected number of days until the certificate becomes valid
+        - this is useful only if you expect the certificate to be valid after a certain date
+        - if missing, 0 is the default value (read more bellow)
+    fail_if_notBefore:
```
Comment (basepi) [inline hubblestack_nova/modules/openssl.py]: I don't love mixing camelCase and underscores. Can we change `notAfter` and `notBefore` to be `not_after` and `not_before`, to make it more pythonic?

File: hubblestack_nova/modules/openssl.py [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+    ret = {'Success': [], 'Failure': [], 'Controlled': []}
+    for tag in __tags__:
+        if fnmatch.fnmatch(tag, tags):
+            for tag_data in __tags__[tag]:
+                if 'control' in tag_data:
+                    ret['Controlled'].append(tag_data)
+                    continue
+
+                endpoint = tag_data.get('endpoint', None)
+                file = tag_data.get('file', None)
```
Comment (basepi) [inline hubblestack_nova/modules/openssl.py]: Can we rename this variable? In python 2, `file` is a builtin function that we shouldn't mask this way.

**Comment by basepi** [comment]
Looks like you included an empty file, runners/ssl_checker.py. Could you remove it?

I also left a couple of inline comments.

File: hubblestack_nova/modules/openssl.py [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+    try:
+        cert = ssl.get_server_certificate((server, port))
+    except Exception:
+        cert = None
+    if not cert:
+        return None
+
+    try:
+        x509 = OpenSSL.crypto.load_certificate(OpenSSL.crypto.FILETYPE_PEM, cert)
+    except OpenSSL.crypto.Error:
```
Comment (basepi) [inline hubblestack_nova/modules/openssl.py]: I wonder if we should log these errors at debug log level or something, so people can more easily track down what's going wrong?

File: hubblestack_nova/modules/openssl.py [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+
+def _load_x509_from_file(cert_file_path):
+    try:
+        cert_file = open(cert_file_path)
+    except IOError:
+        return None
+
+    try:
+        x509 = OpenSSL.crypto.load_certificate(OpenSSL.crypto.FILETYPE_PEM, cert_file.read())
+    except OpenSSL.crypto.Error:
```
Comment (basepi) [inline hubblestack_nova/modules/openssl.py]: Same as above

**Comment by gmiu** [comment]
@basepi, I think these commits are covering all your comments. Let me know if you have any more suggestions about logging.

**Comment by basepi** [comment]
Would it be useful to log the exception as well? Perhaps something like this?

```
except OpenSSL.crypto.Error as exc:
    log.error('Unable to load certificate from {0} into x509 object: {1}'.format(server, exc))
```

**Comment by basepi** [comment]
These changes look great, thanks for making them @avb76!

**Comment by gmiu** [comment]
@basepi, I've modified a bit the logic in order to make the unit tests creation more easy.

**Pull request merged**